### ISSUE TITLE
Query receives Ambiguous field name errror response in GBQ API for php, but executes on Web UI

### ISSUE THREAD

**Issue by GoogleCodeExporter** [runt18/google-bigquery#459]
```
What steps will reproduce the problem?

1.The job listed below was submitted via the GBQ PHP API:
job_xjtrCD5lr6PVqUYEmaiU7Ij1OIs
And receives an error response:
(400)Ambiguous field name 'VISITOR_DIM_ID' in JOIN. Please use the table 
qualifier before field name.

2. The same query was pasted into the Web UI and executes, see job: 
bquijob_5e63814e_1534187f917

What is the expected output? What do you see instead?
I would expect both queries to execute if they contain the same text.

What version of the product are you using? On what operating system?
GBQ PHP API and GBQ Web UI.


```

Original issue reported on code.google.com by `rolandan...@gmail.com` on 4 Mar 2016 at 12:25
- Merged into: #415

**Comment by GoogleCodeExporter** [comment]
```
The difference between the two queries is the one with the error was run with 
"Flatten Results = false".  You can get the web UI to fail the same way if you 
set Flatten under Query Options (which will need a destination and Allow Large 
Results).  So if you are okay with flattened results, you can get this query to 
work by having that open on.  If you need them unflattened, or are curious, 
read on.

Second question, why is it failing.  This is a funny corner case you can get 
into with triple joins.  Here's a minimal reproduction:
SELECT t.a
FROM (SELECT 17 as a, 18 as b) t
JOIN (SELECT 17 as a, 18 as b) u ON t.a = u.a
JOIN (SELECT 18 as foo) v ON u.a = v.foo

and here's an example of how you can rewrite the JOIN source names:
SELECT t.a
FROM (SELECT 17 as a, 18 as b) t
JOIN (
  SELECT a as a_from_u, b as b_from_u 
  FROM (SELECT 17 as a, 18 as b)
) u ON t.a = u.a_from_u
JOIN (SELECT 18 as foo) v ON u.a_from_u = v.foo

Basically that's mucking with the names so they aren't all the same at the 
point they get JOINed.  It's admittedly less than ideal but it may get you 
unblocked, if you do need this to run without flattening.
```

Original comment by `e...@google.com` on 4 Mar 2016 at 9:09
- Changed state: **Accepted**

**Comment by GoogleCodeExporter** [comment]
```
Ah, over here this covers it more thoroughly than I did:

http://stackoverflow.com/questions/28980157/bigquery-same-query-works-when-submi
tted-from-ui-and-reports-sql-syntax-error
```

Original comment by `e...@google.com` on 4 Mar 2016 at 9:11

**Comment by GoogleCodeExporter** [comment]
```
And I also missed that we have a similar issue at #415.  I'll mark this one as 
a duplicate, but please let me know if you have any different problems or the 
workarounds don't work for you.
```

Original comment by `e...@google.com` on 4 Mar 2016 at 9:14
- Changed state: **Duplicate**

**Comment by GoogleCodeExporter** [comment]
```
Thanks for the great answer!
```

Original comment by `rolandan...@gmail.com` on 5 Mar 2016 at 2:15
